bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetLightingHandler.java

```diff
@@ -255,33 +255,43 @@ protected Who getManagedWho() {
         return Who.LIGHTING;
     }
 
+    /**
+     * Handles incoming OpenWebNet messages for this lighting device.
+     * This method processes lighting state changes and propagates APL (Area/Point Light) messages
+     * to appropriate group handlers for hierarchical control.
+     *
+     * @param msg the OpenWebNet message to handle (should be a Lighting message)
+     */
     @Override
     protected void handleMessage(BaseOpenMessage msg) {
         logger.debug("handleMessage({}) for thing: {}", msg, thing.getUID());
         super.handleMessage(msg);
 
+        // Update device state based on thing type - dimmers handle brightness, switches handle on/off
         ThingTypeUID thingType = thing.getThingTypeUID();
         if (THING_TYPE_ZB_DIMMER.equals(thingType) || THING_TYPE_BUS_DIMMER.equals(thingType)) {
             updateBrightness((Lighting) msg);
         } else {
             updateOnOffState((Lighting) msg);
         }
 
+        // Handle APL (Area/Point Light) message propagation for BUS gateway devices
+        // APL messages need to be forwarded to area and general handlers for group control
         OpenWebNetBridgeHandler bridgeHandler = this.bridgeHandler;
         if (bridgeHandler != null && bridgeHandler.isBusGateway()) {
             if (deviceWhere instanceof WhereLightAutom whereLightAutom && ownId != null && whereLightAutom.isAPL()) {
-                // Propagate APL msg to AREA handler, if exists
-                OpenWebNetLightingGroupHandler areaHandler = (OpenWebNetLightingGroupHandler) bridgeHandler
-                        .getRegisteredDevice(areaOwnId);
-                if (areaHandler != null) {
+
+                // First try to propagate to AREA handler (e.g., "1.1" -> area handler for area 1)
+                String areaOwnId = this.areaOwnId;
+                if (areaOwnId != null && bridgeHandler
+                        .getRegisteredDevice(areaOwnId) instanceof OpenWebNetLightingGroupHandler areaHandler) {
```

Comment:
`areaOwnId` was not checked for null and the `getRegisteredDevice` used a list that throws a null when the supplied key is null.